widget/embed_ox_talks.js

```diff
@@ -175,6 +175,115 @@ var oxtalks = {
         return '#' + divname;
     },
     
+    //create a new fullcalendar Eventsource for the given query params, which will be invoked by fullcalendar as the calendar is browsed
+    createEventSource: function(params) {
+        //store the search params for future queries
+        var p = params;
+        
+        //calls fullcalendar's callback with the new set of events
+        var updateCal = function(data, element) {
+            var events = data._embedded.talks;
+            //fullCalendar's callback method should exist in the calling context
+            this.cb(events);            
+        };
+        
+        var events = function(start, end, timezone, callback) {
+            // update query params for new date range
+            p.start = start.format("DD/MM/YY");
+            p.end = end.format("DD/MM/YY");
+            //add the fullcalendar callback to the context
+            this.cb = callback;
+            // fire off the query
+            this.queryTalks(p, updateCal.bind(this), null);
+        };
+        
+        return {
+            events: events.bind(this)    
+            //TODO Customise colours, editability here
+        }
+    },
+    
+    //convert incoming talks json data to the Event format used by fullcalendar
+    ConvertToCalendarEvent: function(talk) {
+        var event = {
+            id: talk.title,
```

Comment:
`id` = `title`?
Not sure what ID is used for but `slug` or `url` might be better identifiers...